Hash_map/include/CGAL/Handle_hash_map.h

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2021  GeometryFactory Sarl (France).
+// All rights reserved.
+//
+// Licensees holding a valid commercial license may use this file in
+// accordance with the commercial license agreement provided with the software.
+//
+// This file is part of CGAL (www.cgal.org)
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: LGPL-3.0-or-later OR LicenseRef-Commercial
+//
+// Author(s)     : Giles Bathgate <giles.bathgate@gmail.com>
+
+#ifndef CGAL_HANDLE_HASH_MAP
+#define CGAL_HANDLE_HASH_MAP
+
+#include <CGAL/config.h>
+#include <CGAL/Handle_hash_function.h>
+#include <CGAL/Tools/robin_hood.h>
+#include <cstddef>
+#include <functional>
+
+namespace CGAL
+{
+namespace internal
+{
+
+template <class Key,
+          class T,
+          class Hash = Handle_hash_function,
+          class KeyEqual = std::equal_to<Key>,
+          class Allocator = void>
+class Handle_hash_map
+  : private robin_hood::unordered_node_map<Key,T,Hash,KeyEqual>
+{
+    typedef robin_hood::unordered_node_map<Key,T,Hash,KeyEqual> Base;
+
+public:
+    // STL compliance
+    typedef Key                           key_type;
+    typedef T                             mapped_type;
+    typedef std::size_t                   size_type;
+    typedef Hash                          hasher;
+    typedef KeyEqual                      key_equal;
+    typedef typename Base::const_iterator const_iterator;
+
+    // Backwards compatibility
+    typedef mapped_type  Data;
+
+private:
+    mapped_type _default;
```

Comment:
@afabri Yes the wrapper handles the default value, and below the code uses try_emplace to utilise this default value.